ompi/mca/pml/ob1/pml_ob1_recvreq.c

```diff
@@ -1124,16 +1128,15 @@ static inline void append_recv_req_to_queue(opal_list_t *queue,
 {
     opal_list_append(queue, (opal_list_item_t*)req);
 
-#if OMPI_WANT_PERUSE
```

Comment:
If nothing is optional, what is the impact on performance for supporting these MPI_T events ?

Reply:
@bosilca I'm assuming that by default we would like the MPI_T Events functions visible, but would like to control whether the implemented events are supported, such as activation with a configure option like with the software performance counters.  Is this correct?

Reply:
@bosilca Here is some preliminary benchmarking I did on the LLNL quartz cluster, comparing OMB MPI microbenchmark times from my PR master branch with the events branch build and running without registered callbacks and then with registered callbacks.  The callbacks do not include reading event data.

Registered callback after MPI_Init for ob1 : ompi_pml_ob1_message_arrived		
Registered callback after MPI_Init for osc : ompi_osc_rdma_get_started		
Reported times are best of 10 runs each.

I'm not sure why the event osc/get timing is so much higher.  I tried to build and test with ompi/main, but ran into PMIx errors.

Are these the results you are interested in?  Let me know if you have specific requests for tests.

<head>

<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:12.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl63
	{text-align:center;
	white-space:normal;}
-->

</head>

<body link="#0563C1" vlink="#954F72">



  | PR Master | Events build, no callbacks | Events w/Registered Callbacks
-- | -- | -- | --
2ppn-1N ob1 pt2pt osu_bw | 3.38 sec | 3.30 sec | 5.09 sec
1ppn-2N ob1 pt2pt osu_bw | 7.08 sec | 7.05 sec | 8.63 sec
  |   |   |  
2ppn-1N osc RMA osu_get_bw | 5.33 sec | 5.00 sec | 6.75 sec
1ppn-2N osc RMA osu_get_bw | 6.14 sec | 7.16 sec | 9.22 sec



</body>

</html>

Reply:
That `event build, no callback`is faster than `no events` is strange and points to some quirks in the measurements. what is the standard deviation of the measurements ?

Reply:
I see a standard deviation of .093 for the 1-node master measurements and .035 for the 1-node event-no-callbacks measurements.  I re-built all of these without debugging and am seeing even more dramatic differences for the measurements from master.  I am building the 4.1.4 release to see if I see better performance.